src/rust-crypto/DehydratedDeviceManager.ts

```diff
@@ -190,7 +202,7 @@ export class DehydratedDeviceManager {
      * Returns whether or not a dehydrated device was found.
      */
     public async rehydrateDeviceIfAvailable(): Promise<boolean> {
-        const key = await this.getKey(false);
+        const key = (await this.getCachedKey()) || (await this.getKey(false));
```

Comment:
Why is `getCachedKey` called, when `getKey` already calls `getCachedKey` and returns its value if it's non-empty?